src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,134 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+
+
+/**
+ * A parser to parse Duke commands.
+ */
+public class Parser {
+    /**
+     * Get the action specified by the command.
+     * <p>
+     * A valid action can be list, done, todo, event, deadline, delete.
+     *
+     * @param command The command to be parsed.
+     * @return The action of the command.
+     */
+    public String getCommandAction(String command) {
+        return command.split(" ", 2)[0];
+    }
+
+    /**
+     * Get the task index of done and delete action.
+     *
+     * @param command The command to be parsed.
+     * @return The index of the task of the command, -1 if is invalid action.
+     */
+    public int getCommandActionIndex(String command) throws DukeException {
+        switch (getCommandAction(command)) {
+        case "done":
+            return Integer.parseInt(command.substring(5));
+        case "delete":
+            return Integer.parseInt(command.substring(7));
+        default:
+            throw new DukeException("Invalid action. Index cannot be parsed from the command.");
+        }
+    }
+
+    /**
+     * Get the relevant task information from the command.
+     *
+     * @param command The command to be parsed.
+     * @return A map of task information from the command.
+     */
+    public Task commandToTask(String command) throws DukeException {
+        switch (getCommandAction(command)) {
+        case "todo":
+            String[] todoSplit = command.split(" ", 2);
+            String todoDescription;
+            try {
+                todoDescription = todoSplit[1];
+            } catch (ArrayIndexOutOfBoundsException e) {
+                // no description
+                String message = "☹ OOPS!!! The description of a todo cannot be empty.";
+                throw new DukeException(message);
+            }
+            return new Todo(todoDescription);
+        case "event":
+            String[] eventSplit = command.split(" /at ");
```

Comment:
(same as todo)
Collections should be represented in plural, so maybe change to eventDetails instead.
```suggestion
            String[] eventDetails = command.split(" /at ");
```